RestaurantReservation.DB/Exceptions/RecordNotFoundException.cs

```diff
@@ -0,0 +1,3 @@
+namespace RestaurantReservation.DB.Exceptions;
+
+public class RecordNotFoundException() : Exception ("Entity not found");
```

Comment:
Maybe consider:

```suggestion
public class RecordNotFoundException : Exception
{
    public RecordNotFoundException() : base("Entity not found")
    {
    }

    public RecordNotFoundException(string message) : base(message)
    {
      
    }
}
```
And then for the BaseRepository:

```
        if (!isExist)
        {
            throw new RecordNotFoundException($"{typeof(T)} with id: {entity.Id} not found");
        }
```

What do you thiink?